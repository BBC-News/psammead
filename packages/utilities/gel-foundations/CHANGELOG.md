# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.4   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Add tests for the exported values, coverage 100% :tada: |
| 0.1.1   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.0   | [PR#221](https://github.com/BBC-News/psammead/pull/221) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |

