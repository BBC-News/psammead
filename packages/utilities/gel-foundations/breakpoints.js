module.exports = require('./dist/breakpoints'); // eslint-disable-line import/no-unresolved
