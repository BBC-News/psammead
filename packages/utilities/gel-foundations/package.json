{
  "name": "drew-testing-123",
  "version": "1.0.6",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "description": "A range of string constants for use in CSS, intended to help implement BBC GEL-compliant webpages and components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/utilities/gel-foundations"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "files": [
    "dist/"
  ],
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "gel",
    "breakpoints",
    "grid",
    "spacing",
    "typography"
  ],
  "devDependencies": {
    "@bbc/psammead-test-helpers": "^1.0.1"
  }
}
