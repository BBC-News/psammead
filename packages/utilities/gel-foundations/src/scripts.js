export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
