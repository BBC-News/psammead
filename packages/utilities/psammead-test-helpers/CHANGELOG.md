# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.3.0   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Adds a helper method for testing package export values and uses it to test its own exports. |
| 0.2.1   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
