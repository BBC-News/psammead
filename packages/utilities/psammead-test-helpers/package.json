{
  "name": "@bbc/psammead-test-helpers",
  "version": "0.3.0",
  "main": "dist/index.js",
  "description": "A collection of helper methods for implementing Jest snapshot tests for styled-components, required by many Psammead components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/BBC-News/psammead/tree/latest/packages/utilities/psammead-test-helpers"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/BBC-News/psammead/blob/latest/packages/utilities/psammead-test-helpers/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "test",
    "helpers",
    "jest",
    "snapshot"
  ],
  "dependencies": {
    "jest-styled-components": "^6.3.1",
    "react-test-renderer": "^16.6.3"
  }
}
