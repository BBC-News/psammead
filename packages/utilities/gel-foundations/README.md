# gel-foundations &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/bbc/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/gel-foundations.svg)](https://www.npmjs.com/package/@bbc/gel-foundations) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md)

This package provides a range of string constants for use in CSS, intended to help implement [BBC GEL-compliant](https://www.bbc.co.uk/gel/articles/what-is-gel) webpages and components.

## Exports

`/breakpoints` - GEL breakpoints, as well as typography breakpoints. These use the GEL grid sizes which can be found in the [GEL Grid guidelines](https://www.bbc.co.uk/gel/guidelines/grid#grid-sizes).  
`/spacings` - GEL spacings and GEL Grid margins and gutters. These use the GEL grid spacing sizes which can be found in the [GEL Grid guidelines](https://www.bbc.co.uk/gel/guidelines/grid#spacing-layout).  
`/typography` - GEL typography. These are based on the gel typography standard which can be found in the [GEL guidelines](https://www.bbc.co.uk/gel/guidelines/typography). [More details on how these sizes were implemented for this package are available here.](./typography_sizes_web.md)

## Installation

```jsx
npm install @bbc/gel-foundations --save
```

## Usage

```jsx
import { GEL_GROUP_3_SCREEN_WIDTH_MIN } from '@bbc/gel-foundations/breakpoints';

import { GEL_GUTTER_ABOVE_600PX } from '@bbc/gel-foundations/spacings';

import { GEL_BREVIER } from '@bbc/gel-foundations/typography';
```

These values can then be used directly within CSS declarations in code:

```jsx
import { css } from 'styled-components';

const SomeStyledComponent = css`
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    grid-gap: ${GEL_GUTTER_ABOVE_600PX};
    ${GEL_BREVIER};
  }
`;
```

To allow the typography to be fully accessible and responsive, please note that you should apply a default font-size to the document root (e.g. `html { font-size: 100% }`).

Our typography uses `em` for font-size and `rem` for line-height. `em` allows modularity of components: you can change the component font-size by changing the font-size of its container. `rem` is relative to the document root, so we use that for line-height and spacing for a consistent look-and-feel across the document. You can read our [detailed analysis of "REMs vs EMs for spacing"](https://github.com/bbc/simorgh/blob/latest/docs/Spacing-Units.md) for more information.

## Contributing

When **adding** a new export to this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/gel-foundations/index.test.jsx#L13-L59) also need to be updated. When **removing** an exisiting export from this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/gel-foundations/index.test.jsx#L13-L59) need to be updated and the package version requires a major change (EG: 1.2.1 -> 2.0.0) as this would be considered a breaking change as functionality is being removed.

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/bbc/psammead/blob/latest/LICENSE).
