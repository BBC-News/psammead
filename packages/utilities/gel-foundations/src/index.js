import * as breakpoints from './breakpoints';
import * as propTypes from './prop-types';
import * as scripts from './scripts';
import * as spacings from './spacings';
import * as typography from './typography';

export { breakpoints };
export { propTypes };
export { scripts };
export { spacings };
export { typography };
