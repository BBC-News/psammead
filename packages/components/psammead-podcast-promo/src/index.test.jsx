import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import * as scripts from '@bbc/gel-foundations/scripts';

import PodcastPromo from '.';
import BasicExample from './examples/basic';
import OnPageExample from './examples/on-page';

const assertTypeOfElement = (Component, type) => {
  const { container } = render(
    <Component script={scripts.latin} service="russian">
      Content
    </Component>,
  );

  expect(container.querySelector(type)).toBeInTheDocument();
  expect(container.querySelector(type).textContent).toBe('Content');
};

describe('Podcast Promo', () => {
  describe('Title', () => {
    assertTypeOfElement(PodcastPromo.Title, 'h2');
    shouldMatchSnapshot(
      'should match snapshot',
      <PodcastPromo.Title script={scripts.latin} service="russian">
        Content
      </PodcastPromo.Title>,
    );
  });

  describe('Card', () => {
    assertTypeOfElement(PodcastPromo.Title, 'div');
    shouldMatchSnapshot(
      'should match snapshot',
      <PodcastPromo.Card>Content</PodcastPromo.Card>,
    );
  });

  describe('Card Content', () => {
    assertTypeOfElement(PodcastPromo.Card.Content, 'div');
    shouldMatchSnapshot(
      'should match snapshot',
      <PodcastPromo.Card.Content>Content</PodcastPromo.Card.Content>,
    );
  });

  describe('Card Title', () => {
    assertTypeOfElement(PodcastPromo.Card.Title, 'h3');
    shouldMatchSnapshot(
      'should match snapshot',
      <PodcastPromo.Card.Title script={scripts.latin} service="russian">
        Content
      </PodcastPromo.Card.Title>,
    );
  });

  describe('Card Description', () => {
    assertTypeOfElement(PodcastPromo.Card.Description, 'p');
    shouldMatchSnapshot(
      'should match snapshot',
      <PodcastPromo.Card.Description script={scripts.latin} service="russian">
        Content
      </PodcastPromo.Card.Description>,
    );
  });

  describe('Card Image', () => {
    assertTypeOfElement(PodcastPromo.Card.Image, 'figure');
    shouldMatchSnapshot(
      'should match snapshot',
      <PodcastPromo.Card.Image>Content</PodcastPromo.Card.Image>,
    );
  });

  describe('Card Link', () => {
    assertTypeOfElement(PodcastPromo.Card.Link, 'a');
    shouldMatchSnapshot(
      'should match snapshot',
      <PodcastPromo.Card.Link href="https://www.bbc.com">
        Content
      </PodcastPromo.Card.Link>,
    );
  });

  describe('card-call-to-action', () => {
    assertTypeOfElement(PodcastPromo.Card.CallToAction, 'span');
    shouldMatchSnapshot(
      'should match snapshot',
      <PodcastPromo.Card.CallToAction script={scripts.latin} service="russian">
        Click Me
      </PodcastPromo.Card.CallToAction>,
    );
    it('should be aria-hidden', () => {
      const { container } = render(
        <PodcastPromo.Card.CallToAction
          script={scripts.latin}
          service="russian"
        >
          Click Me
        </PodcastPromo.Card.CallToAction>,
      );

      expect(container.querySelector('span')).toHaveAttribute(
        'aria-hidden',
        'true',
      );
    });
  });

  describe('Examples', () => {
    shouldMatchSnapshot(
      'basic example',
      <BasicExample script={scripts.latin} service="russian" />,
    );
    shouldMatchSnapshot(
      'on-page example',
      <OnPageExample script={scripts.latin} service="russian" />,
    );
  });
});
