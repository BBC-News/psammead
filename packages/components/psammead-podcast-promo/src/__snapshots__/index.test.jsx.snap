// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Podcast Promo Card Content should match snapshot 1`] = `
.emotion-0 {
  padding: 0.5rem;
  display: inline-block;
  max-width: calc(100% - 96px);
}

@media (min-width:25rem) {
  .emotion-0 {
    padding: 0.5rem 1rem;
    max-width: calc(100% - 117px);
  }
}

@media (min-width:37.5rem) {
  .emotion-0 {
    padding: 0.5rem 1rem;
    max-width: calc(100% - 178px);
  }
}

@media (min-width:63rem) {
  .emotion-0 {
    padding: 1rem;
    max-width: initial;
  }
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    Content
  </div>
</div>
`;

exports[`Podcast Promo Card Description should match snapshot 1`] = `
.emotion-0 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  max-width: 30rem;
  color: #6E6E73;
  margin-top: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-0 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-0 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

<div>
  <p
    class="emotion-0 emotion-1"
  >
    Content
  </p>
</div>
`;

exports[`Podcast Promo Card Image should match snapshot 1`] = `
.emotion-0 {
  width: 88px;
  display: inline-block;
  vertical-align: top;
  margin: 8px 0 0 8px;
}

@media (min-width:25rem) {
  .emotion-0 {
    width: 109px;
  }
}

@media (min-width:37.5rem) {
  .emotion-0 {
    width: 178px;
    margin: 0;
  }
}

@media (min-width:63rem) {
  .emotion-0 {
    width: 100%;
    margin: 0;
  }
}

<div>
  <figure
    class="emotion-0 emotion-1"
  >
    Content
  </figure>
</div>
`;

exports[`Podcast Promo Card Link should match snapshot 1`] = `
.emotion-0 {
  color: unset;
  -webkit-text-decoration: unset;
  text-decoration: unset;
}

.emotion-0:before {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  content: '';
  overflow: hidden;
  z-index: 1;
}

.emotion-0:visited .podcast-promo--visited {
  color: #AEAEB5;
}

.emotion-0:focus .podcast-promo--focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <a
    class="emotion-0 emotion-1"
    href="https://www.bbc.com"
  >
    Content
  </a>
</div>
`;

exports[`Podcast Promo Card Title should match snapshot 1`] = `
.emotion-0 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  margin-top: 0;
  margin-bottom: 0.5rem;
  color: #222222;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

<div>
  <h3
    class="emotion-0 emotion-1"
  >
    Content
  </h3>
</div>
`;

exports[`Podcast Promo Card should match snapshot 1`] = `
.emotion-0 {
  position: relative;
  background-color: #FFFFFF;
}

.emotion-0:hover .podcast-promo--hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    Content
  </div>
</div>
`;

exports[`Podcast Promo Title should match snapshot 1`] = `
.emotion-6 {
  margin: 0 0 1rem;
}

.emotion-2 > svg {
  margin-left: 0;
  width: 22px;
  height: 22px;
  fill: currentColor;
  position: relative;
  bottom: 5px;
  right: 3px;
}

.emotion-0 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 1rem;
  height: 1rem;
}

.emotion-4 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  display: inline;
  color: #222222;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-4 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-4 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

<div>
  <div
    class="emotion-6 emotion-7"
  >
    <i
      class="emotion-2 emotion-3"
    >
      <svg
        aria-hidden="true"
        class="emotion-0 emotion-1"
        focusable="false"
        height="32"
        viewBox="0 0 32 32"
        width="32"
      >
        <path
          d="M18.7,31h-5.3l-2.3-10.4C12.3,19.5,14,19,16,19s3.7,0.5,4.9,1.7L18.7,31z M22,8.2c-1.7-1.7-3.9-2.5-6.1-2.5s-4.4,0.9-6,2.5 l1.7,1.7c1.2-1.2,2.7-1.8,4.3-1.8s3.1,0.6,4.3,1.8L22,8.2z M25.5,4.9c-2.6-2.7-6.1-4-9.5-4S9.1,2.3,6.5,4.9l1.7,1.7 c2.1-2.1,4.9-3.2,7.7-3.2c2.8,0,5.6,1.1,7.8,3.2L25.5,4.9z M12.4,14c0,2,1.5,3.6,3.6,3.6c2,0,3.6-1.5,3.6-3.6 c0-2.1-1.5-3.6-3.6-3.6C13.9,10.4,12.4,11.9,12.4,14z"
        />
      </svg>
    </i>
    <h2
      class="emotion-4 emotion-5"
    >
      Content
    </h2>
  </div>
</div>
`;

exports[`Podcast Promo card-call-to-action should match snapshot 1`] = `
.emotion-2 > svg {
  width: 16px;
  height: 16px;
  fill: currentColor;
  position: relative;
  bottom: 2px;
  right: 3px;
}

.emotion-0 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 1rem;
  height: 1rem;
}

.emotion-4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #6E6E73;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .emotion-4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .emotion-4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

<div>
  <span
    aria-hidden="true"
  >
    <i
      class="emotion-2 emotion-3"
    >
      <svg
        aria-hidden="true"
        class="emotion-0 emotion-1"
        focusable="false"
        height="32"
        viewBox="0 0 32 32"
        width="32"
      >
        <polygon
          points="4 6 11.1 6 26 6 26 28 28 28 28 4 4 4 4 6"
        />
        <polygon
          points="8 0 8 2 30 2 30 24 32 24 32 0 8 0"
        />
        <path
          d="M0,32H24V8H0ZM4,12H20V28H4Z"
        />
      </svg>
    </i>
    <span
      class="emotion-4 emotion-5"
    >
      Click Me
    </span>
  </span>
</div>
`;
